tools/cassandra_delete_range/main.go

```diff
@@ -424,30 +477,43 @@ func prepareDeleteQueries(cluster *gocql.ClusterConfig, fromLedgerIdx uint64, qu
 
 				sessionCreationWaitGroup.Done()
 				sessionCreationWaitGroup.Wait()
-				preparedQueryString := fmt.Sprintf(queryTemplate, "?", "?")
-				preparedQuery := session.Query(preparedQueryString)
+				preparedQuery := session.Query(q)
 
 				for r := range rangesChannel {
 					preparedQuery.Bind(r.StartRange, r.EndRange)
-					iter := preparedQuery.Iter()
 
-					var key []byte
-					var seq uint64
+					var pageState []byte
 					var rowsRetrieved uint64
 
-					for iter.Scan(&key, &seq) {
-						rowsRetrieved++
+					for {
+						iter := preparedQuery.PageSize(*clusterPageSize).PageState(pageState).Iter()
+						nextPageState := iter.PageState()
+						scanner := iter.Scanner()
+
+						for scanner.Next() {
+							var key []byte
```

Comment:
If, for the same query, the keys are guaranteed to be the same size, the it should be safe to move `key` and `seq` outside of the loop stack.